action.yml

```diff
@@ -7,31 +7,30 @@ runs:
       shell: bash
       run: |
         rm -rf '${{github.workspace}}'
-    - name: Clone
+    - name: Init
       shell: bash
       working-directory: '${{runner.workspace}}'
       run: |
-        git clone 'git@github.com:${{github.repository}}'
-    - name: Checkout for PRs
+        git init '${{github.workspace}}'
+    - name: Add origin
       shell: bash
       working-directory: '${{github.workspace}}'
-      if: ${{ github.event_name == 'pull_request' }}
       run: |
-        git checkout '${{github.event.pull_request.head.sha}}'
-        git config user.name "Actions Runner"
-        git config user.email "noreply@example.com"
-        git merge --no-commit 'origin/${{github.base_ref}}'
```

Comment:
I think this will lead to problems. The way I wrote it, master will automatically be merged into each PR. Without this, out-of-date branches could pass CI but then implode when combined with the latest master code upon merging the PR.

Reply:
There's some details in [06ffd36](https://github.com/TransitApp/action-checkout/pull/1/commits/06ffd36d55ffdbd9dcc6707040fb4067e54491d7) ... I think we will have master's commits this way

Reply:
Yeah I definitely see what you mean based on the documents you linked, however I'm still worried it'll introduce subtle bugs. For instance, I have tried to use `github.sha` in the past for this action but had to revert it (e.g. 58862a). Is it possible to merge only the switch to shallow clones for now, to minimize the risk of this PR breaking things?

Reply:
https://github.com/TransitApp/action-checkout/commit/58862ab0c9ea58b3178fb93da281f90ba5baf0ba

Reply:
It's this line from before that I'm concerned about keeping:
```
        git merge --no-commit 'origin/${{github.base_ref}}'
```

I think I can support it by doing a fetch of both top that it wants to merge, but I don't see how the `stefanzweifel/git-auto-commit-action` can behave reliably if the checkout has uncommitted changes from a merge in it. Are you using that action in combination with this action already?

Reply:
No we don't currently use that combination. `filePattern` would help unless that particular file also has changes from master.

Reply:
I wonder if we should deploy your changes now, with the expectation that we might have to revert if we notice unexpected behaviour on any PRs?

Reply:
That's cool with me, if we need to revert I'll jump on implementing an explicit merge as an option. But this really ought to work, it's similar to what github's own checkout action is doing 🙏